changelog/5515.feature.rst

```diff
@@ -0,0 +1,15 @@
+Allow selective auto-indentation of multiline log messages.
+
+Supports command line option ``--log-auto-indent [value]``
+and config option ``log_auto_indent`` to set the default for all logging.
+
+Supports passing kwarg ``extra={"auto_indent": [value]}`` to
```

Comment:
What is the motiviation for adding an auto_indent parameter to `log.log`? Do you want to configure auto_indent per log record?

Reply:
yes, this adds per log line support, so that you can strategically turn it off when printing a preformatted table of trace info, for example.

Reply:
@earonesty is correct. Sometimes it is good to leave it default on, and then strategically turn it off on a particularly long log entry that doesn't want to be pushed so far to the right, or to set the indentation position to a position that is suited to that entry